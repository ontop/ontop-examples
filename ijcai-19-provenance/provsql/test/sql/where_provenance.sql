\set ECHO none
SET search_path TO public, provsql;

/* Example of where-provenance */
CREATE TABLE result_where AS
  SELECT p1.city AS c1, p2.city AS c2,
    regexp_replace(where_provenance(provenance()),':[0-9a-f-]*:','::','g')
  FROM personnel p1, personnel p2
  WHERE p1.city = p2.city AND p1.id < p2.id
  GROUP BY p1.city, p2.city
  ORDER BY p1.city;

SELECT remove_provenance('result_where');
SELECT * FROM result_where;
DROP TABLE result_where;

CREATE TABLE result_where AS
  SELECT city,
    regexp_replace(where_provenance(provenance()),':[0-9a-f-]*:','::','g')
  FROM (
    SELECT DISTINCT p1.city 
    FROM personnel p1, personnel p2 
    WHERE p2.city='Paris'
  ) t
  ORDER BY city;

SELECT remove_provenance('result_where');
SELECT * FROM result_where;
DROP TABLE result_where;

CREATE TABLE result_where AS
  SELECT 1,
  city,
  regexp_replace(where_provenance(provenance()),':[0-9a-f-]*:','::','g')
  FROM (
    SELECT city
    FROM personnel
  ) t
  ORDER BY city;

SELECT remove_provenance('result_where');
SELECT * FROM result_where;
DROP TABLE result_where;
