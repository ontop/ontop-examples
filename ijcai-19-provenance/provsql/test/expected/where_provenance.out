\set ECHO none
 remove_provenance 
-------------------
 
(1 row)

    c1    |    c2    |                                     regexp_replace                                     
----------+----------+----------------------------------------------------------------------------------------
 Berlin   | Berlin   | {[personnel::4;personnel::4],[personnel::4;personnel::4],[]}
 New York | New York | {[personnel::4;personnel::4],[personnel::4;personnel::4],[]}
 Paris    | Paris    | {[personnel::4;personnel::4;personnel::4],[personnel::4;personnel::4;personnel::4],[]}
(3 rows)

 remove_provenance 
-------------------
 
(1 row)

   city   |                regexp_replace                 
----------+-----------------------------------------------
 Berlin   | {[personnel::4;personnel::4],[]}
 New York | {[personnel::4;personnel::4],[]}
 Paris    | {[personnel::4;personnel::4;personnel::4],[]}
(3 rows)

 remove_provenance 
-------------------
 
(1 row)

 ?column? |   city   |     regexp_replace     
----------+----------+------------------------
        1 | Berlin   | {[],[personnel::4],[]}
        1 | Berlin   | {[],[personnel::4],[]}
        1 | New York | {[],[personnel::4],[]}
        1 | New York | {[],[personnel::4],[]}
        1 | Paris    | {[],[personnel::4],[]}
        1 | Paris    | {[],[personnel::4],[]}
        1 | Paris    | {[],[personnel::4],[]}
(7 rows)

